refpolicy: modify kerberos.te

We need to have write access to a file under /var,
and kadmind needs to be able to bind to more
than just the kerberos-defined ports.  Allowing it
to bind to all reserved ports is a bit lax, but there
is no other interface that gets us what we need.

Upstream-Status: Pending

Signed-off-by: Joe Slater <jslater@windriver.com>


--- a/policy/modules/contrib/kerberos.te
+++ b/policy/modules/contrib/kerberos.te
@@ -123,7 +123,8 @@ corenet_udp_bind_generic_node(kadmind_t)
 corenet_sendrecv_all_server_packets(kadmind_t)
 corenet_tcp_bind_kerberos_admin_port(kadmind_t)
 corenet_udp_bind_kerberos_admin_port(kadmind_t)
-corenet_tcp_bind_reserved_port(kadmind_t)
+corenet_tcp_bind_kerberos_master_port(kadmind_t)
+corenet_tcp_bind_all_reserved_ports(kadmind_t)
 
 dev_read_sysfs(kadmind_t)
 
@@ -132,9 +133,11 @@ fs_search_auto_mountpoints(kadmind_t)
 
 domain_use_interactive_fds(kadmind_t)
 
+files_read_all_symlinks(kadmind_t)
+
 files_read_etc_files(kadmind_t)
 files_read_usr_files(kadmind_t)
-files_read_var_files(kadmind_t)
+files_rw_var_files(kadmind_t)
 
 selinux_validate_context(kadmind_t)
 
@@ -238,9 +241,10 @@ fs_search_auto_mountpoints(krb5kdc_t)
 
 domain_use_interactive_fds(krb5kdc_t)
 
+files_read_all_symlinks(krb5kdc_t)
+
 files_read_etc_files(krb5kdc_t)
-files_read_usr_symlinks(krb5kdc_t)
-files_read_var_files(krb5kdc_t)
+files_rw_var_files(krb5kdc_t)
 
 selinux_validate_context(krb5kdc_t)
 
